File: 88-merge-sorted-array/merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int i = 0, j = 0,c=0;
        int a[max(m,1)];
        for (int k = 0;k<m;k++){
            a[k]=nums1[k];
        }
        while(i<m & j<n){
            nums1[c++] = a[i]<nums2[j]?a[i++]:nums2[j++];
        }
        while(i<m){
            nums1[c++] = a[i++];
        }
        while(j<n){
            nums1[c++] = nums2[j++];
        }
    }
};

File: 88-merge-sorted-array/merge-sorted-array.py
class Solution:
    def merge(self, a: List[int], m: int, b: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        i = m - 1
        j = n - 1
        c = m + n -1
        while (i >= 0 and j >= 0):
            if a[i] > b[j]:
                a[c] = a[i]
                i -= 1
            else:
                a[c] = b[j]
                j -= 1
            c -= 1
        while j >= 0:
            a[c] = b[j]
            j -= 1
            c -= 1
            